tools_docker/tracim_debian_uwsgi/Dockerfile

```
@@ -0,0 +1,183 @@
+
+ # Clone repo
+FROM base as base_repository_local
+ARG LOCAL_REPO_DIR
+RUN if [ "$LOCAL_REPO_DIR" = "" ]; then echo "No local repository dir LOCAL_REPO_DIR given." && exit 2; fi
+ADD ${LOCAL_REPO_DIR} /tracim
+
+FROM base as base_repository_remote
+ARG TAG
+ARG BRANCH
+ARG REPO
+RUN if ([ "$TAG" != "" ] && [ "$BRANCH" != "" ]); \
+    then echo "Its not possible to use ARG TAG and ARG BRANCH in same time." && exit 1;fi
+RUN \
+    get_source_from_github() { \
+    if ([ "$TAG" != "" ] && [ "$BRANCH" = "" ]); \
+    then git clone -b $TAG --single-branch "$REPO" tracim; \
+    elif ([ "$TAG" = "" ] && [ "$BRANCH" != "" ]); \
+    then git clone "$REPO" tracim && cd tracim/ && git checkout $BRANCH && cd -; \
+    else git clone "$REPO" tracim; \
+    fi \
+    } \
+    && get_source_from_github $TAG $BRANCH
+FROM base_repository_${REPOSITORY_TYPE} as base_repository
+
+FROM base_repository as base_install
+ARG PACKAGE_DIR
+RUN cat $PACKAGE_DIR/base_packages.list > packages.list
+RUN cat $PACKAGE_DIR/pg_minimal_packages.list >> packages.list
+RUN if [ "$DEBUG_DEP" = "full" ]; then cat $PACKAGE_DIR/debug_packages.list >> packages.list ; fi
+RUN if [ "$ENCRYPTION_DEP" = "full" ]; then cat $PACKAGE_DIR/encryption_packages.list >> packages.list ; fi
+RUN if [ "$PG_DEP" = "full" ]; then cat $PACKAGE_DIR/pg_full_packages.list >> packages.list ; fi
+RUN \
+    apt update \
+    && apt upgrade -qy \
+    && xargs -a packages.list apt install -qy --no-install-recommends\
+    && apt autoremove -qy \
+    && rm -rf /var/lib/{apt,dpkg,cache,log}/ /tmp/* /var/tmp/*
+
+
+
+
+###
+# Backend install stage
+####
+# Stage who:
+# - install python dependencies
+# - install tracim backend
+# - install optionals dependencies
+####
+
+# level 1 : base builder
+FROM base_install AS base_backend
+ARG PACKAGE_DIR
+# Install required packages
+RUN xargs -a $PACKAGE_DIR/install_backend_packages.list apt install -qy
+RUN \
+# Go in backend folder
+    cd /tracim/backend \
+# Install backend
+    && python3 -m pip install -r requirements-build.txt \
+    && python3 -m pip install -r requirements.txt \
+    && python3 -m pip install -e "."
+
+# level 2 : db requirement stage
+FROM base_backend as db_install_minimal
+FROM base_backend as db_install_full
+RUN \
+    # Go in backend folder
+    cd /tracim/backend \
+    && python3 -m pip install -r requirements-db-postgres.txt \
+    && python3 -m pip install -r requirements-db-mysql.txt
+FROM db_install_${DB_DEP} as db_install
+
+# level 3 : PG dependencies stage
+FROM db_install as pg_install_minimal
+FROM db_install as pg_install_full
+RUN \
+    # Go in backend folder
+    cd /tracim/backend \
+    && python3 -m pip install -r requirements-full-preview-generator.txt
+# end of all stages:
+from pg_install_${PG_DEP} as backend
+
+
+###
+# Frontend build stage
+####
+# Stage who:
+# - install javascript dependencies
+# - setup backend translation
+# - build frontend
+####
+FROM node:12-buster AS frontend
```

```suggestion
FROM node:14-buster AS frontend
```
use node 14 to be consistent with https://github.com/tracim/tracim/issues/4830 change.